src/store/Disks.cc

```diff
@@ -363,17 +375,27 @@ Store::Disks::maxObjectSize() const
 }
 
 void
-Store::Disks::updateLimits()
+Store::Disks::configure()
 {
+    if (!Config.cacheSwap.swapDirs)
+        Controller::store_dirs_rebuilding = 0; // nothing to index
+
     largestMinimumObjectSize = -1;
     largestMaximumObjectSize = -1;
     secondLargestMaximumObjectSize = -1;
 
     for (int i = 0; i < Config.cacheSwap.n_configured; ++i) {
-        const auto &disk = Dir(i);
+        auto &disk = Dir(i);
+        if (disk.needsDiskStrand()) {
+            assert(InDaemonMode());
+            disk.disker = Config.workers + (++Config.cacheSwap.n_strands);
+        }
+
         if (!disk.active())
             continue;
 
+        // update limits
+
```

Comment:
If the overall shape of this method remains intact, then please drop this comment: The code updating the limits starts earlier that this comment implies (before the loop). We probably cannot accurately mark configuration and limits sections without segregating/splitting the code using two loops.